lib/network.py

```diff
@@ -27,6 +28,38 @@
 CA_PEM = os.getenv("COCKPIT_CA_PEM", os.path.join(IMAGES_DIR, "files", "ca.pem"))
 
 
+CA_PEM_DOMAINS = [
+    ".e2e.bos.redhat.com",
+    # points to our AWS sink; obsolete after 2022-07 when we settled down on "S3 logs only"
+    "logs.cockpit-project.org",
+    # development/cockpituous project tests
+    ".localdomain",
+]
+
+
+def get_host_ca(hostname):
+    '''Return custom CA that applies to the given host name.
+
+    Self-hosted infrastructure uses CA_PEM, while publicly hosted infrastructure ought to have
+    an officially trusted TLS certificate. Return None for these.
+    '''
+    # strip off port
+    hostname = hostname.split(':')[0]
+
+    if any(hostname.endswith(domain) for domain in CA_PEM_DOMAINS):
```

Comment:
This isn't exactly correct.  It would also match `blogs.cockpit-project.org`, for example.

I'm willing to let it slide, though :)

Reply:
This nags me too much to let it slip -- I had it correct the first time, and then oversimplified it. Fixed.